src/App.tsx

```diff
@@ -14,11 +16,111 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type Props = {};
+
+type State = {
+  selectedGood: string[];
+};
+
+class App extends React.Component<Props, State> {
+  state: State = {
+    selectedGood: ['Jam'],
+  };
+
+  selectHandler = (good: string) => {
+    this.setState((prevState) => {
+      const newState = [...prevState.selectedGood];
+
+      if (prevState.selectedGood.includes(good)) {
+        newState.splice(newState.indexOf(good), 1);
```

Comment:
you iterate through array twice - indexOf and splice.
maybe better to use filter in this case